core/vertx/src/main/java/io/thorntail/vertx/tracing/VertxMessageSpanHandler.java

```diff
@@ -0,0 +1,64 @@
+package io.thorntail.vertx.tracing;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.interceptor.InvocationContext;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.propagation.Format;
+import io.opentracing.tag.Tags;
+import io.thorntail.tracing.SpanHandler;
+import io.thorntail.vertx.VertxExtension;
+import io.thorntail.vertx.VertxLogger;
+import io.thorntail.vertx.VertxMessage;
+
+/**
+ * Starts span for traced {@link VertxMessage} observers.
+ *
+ * @author Martin Kouba
+ */
+@Priority(1)
+@ApplicationScoped
+public class VertxMessageSpanHandler extends SpanHandler {
+
+    static final String OP_NAME_PREFIX = "vertx-observer-notify";
+
+    @Inject
+    private VertxExtension vertExtension;
+
+    @Inject
+    private Tracer tracer;
+
+    @Override
+    public boolean canHandle(InvocationContext ctx) {
+        return vertExtension.isVertxMessageObserver(ctx.getMethod());
+    }
+
+    @Override
+    public ActiveSpan handle(InvocationContext ctx) {
+        VertxMessage vertxMessage = null;
+        for (Object param : ctx.getParameters()) {
+            if (param instanceof VertxMessage) {
+                vertxMessage = (VertxMessage) param;
+                break;
+            }
+        }
+        if (vertxMessage == null) {
+            return null;
+        }
+        VertxLogger.LOG.startSpanForVertxMessageObserver(ctx.getMethod());
+        SpanContext parent = tracer.extract(Format.Builtin.HTTP_HEADERS, new VertxMessageTextMap(vertxMessage));
+        SpanBuilder builder = tracer.buildSpan(String.format("%s:%s.%s", OP_NAME_PREFIX, ctx.getMethod().getDeclaringClass().getName(), ctx.getMethod().getName()));
+        builder.withTag(Tags.MESSAGE_BUS_DESTINATION.getKey(), vertxMessage.address());
+        builder.withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CONSUMER);
+        if (parent != null) {
+            builder.asChildOf(parent);
+        }
+        return builder.startActive();
```

Comment:
I am not sure whether this is correct. (also related to https://github.com/thorntail/thorntail/pull/1024/files#diff-d31eddb96f67822dfa6f43f5f08c4a01R37) 

I am not sure what the vert.x integration in swarm is, but in vertx you cannot rely on active span because it's stored in thread locals.

Reply:
I think this should work even if using the event loop thread  because `VertxMessageSpanHandler` is only used to start an active span for a CDI observer notification (span is deactivated once the invocation completes).